src/amqp.rs

```diff
@@ -0,0 +1,129 @@
+use lapin::tcp::{OwnedIdentity, OwnedTLSConfig};
+use serde::{Deserialize, Serialize};
+use std::path::PathBuf;
+use std::time::Duration;
+use tokio_amqp::*;
+
+/// Client certificate for rabbit authentication
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq)]
+pub(crate) enum ClientCertDer {
+    /// Certificate embedded in config file as base64 string
+    Embedded(String),
+    /// Certificate on file system
+    Path(PathBuf),
+}
+
+/// Tls identity configuration parameters
+#[derive(Clone, Debug, Deserialize, Serialize, Eq, PartialEq)]
+pub(crate) struct TlsIdentity {
+    /// Client certificate and key
+    pub client_cert_and_key_der: ClientCertDer,
+    /// Password for client cert and key
+    #[serde(default = "crate::amqp::tls_identity::client_cert_and_key_password_default")]
```

Comment:
Seems like the default here is just `""`. Do we need the separate module / function? That should be the `Default` for `String` already.